packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -115,203 +129,284 @@ async function setup(getTestObjectProvider, apis) {
-	// 		);
-
-	// 		const rootNode = treeRoot.currentView.getViewNode(treeRoot.currentView.root);
-	// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-	// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
-	// 		const change: Change = Change.setPayload(nodeId, handle);
-	// 		treeRoot.applyEdit(change);
-	// 	},
-	// },
 	{
-		type: registerId,
-		storeHandle: async (defaultDataStore, handle) => {
-			const registerRoot =
-				await defaultDataStore.getSharedObject<IConsensusRegisterCollection<FluidObject>>(
-					registerId,
-				);
-			await registerRoot.write("B", handle);
+		id: treeId,
+		type: SharedTree.getFactory().type,
+		createDDS(runtime, apis) {
+			const tree = runtime.createChannel(undefined, SharedTree.getFactory().type);
+			return this.downCast(tree);
 		},
-		readHandle: async (defaultDataStore) => {
-			const registerRoot =
-				await defaultDataStore.getSharedObject<IConsensusRegisterCollection<FluidObject>>(
-					registerId,
-				);
-			return registerRoot.read("B");
+		downCast(channel): aDDSType {
+			const tree = channel as ISharedTree;
+			const treeView = treeSetup(tree);
+
+			return {
+				id: tree.id,
+				storeHandle(handle: IFluidHandle) {
+					treeView.root.h = handle;
+				},
+				async readHandle(): Promise<unknown> {
+					return treeView.root.h;
+				},
+				handle: tree.handle,
+			};
+		},
+		async getDDS(dataStore) {
+			const tree = await dataStore.getSharedObject<ISharedTree>(treeId);
+			return this.downCast(tree);
 		},
 	},
 	{
-		type: queueId,
-		storeHandle: async (defaultDataStore, handle) => {
-			const queueRoot =
-				await defaultDataStore.getSharedObject<IConsensusOrderedCollection>(queueId);
-			await queueRoot.add(handle);
+		id: registerId,
+		type: ConsensusRegisterCollectionFactory.Type,
+		createDDS(runtime, apis) {
+			const { ConsensusRegisterCollection } = apis.dds;
+			const register = runtime.createChannel(
+				undefined,
+				ConsensusRegisterCollection.getFactory().type,
+			);
+			return this.downCast(register);
 		},
-		readHandle: async (defaultDataStore) => {
-			let handle2: unknown;
-			const queueRoot =
-				await defaultDataStore.getSharedObject<IConsensusOrderedCollection>(queueId);
-			const callback: ConsensusCallback<IFluidHandle> = async (value) => {
-				handle2 = value;
-				return ConsensusResult.Complete;
+		downCast(channel): aDDSType {
+			const register = channel as IConsensusRegisterCollection<FluidObject>;
+			return {
+				id: register.id,
+				storeHandle(handle: IFluidHandle) {
+					void register.write("B", handle);
+				},
+				async readHandle(): Promise<unknown> {
+					return register.read("B");
+				},
+				handle: register.handle,
 			};
-			await queueRoot.waitAndAcquire(callback);
-			return handle2;
+		},
+		async getDDS(dataStore) {
+			const register =
+				await dataStore.getSharedObject<IConsensusRegisterCollection<FluidObject>>(
+					registerId,
+				);
+			return this.downCast(register);
 		},
 	},
 	// {
```

Comment:
ConsensusQueue has an issue that causes the tests to hang. I'm looking into it, but commented it out for now.